extensions/src/AWSSDK.Extensions.NETCore.Setup/ClientFactory.cs

```diff
@@ -135,13 +134,16 @@ private static void PerformGlobalConfig(ILogger logger, AWSOptions options)
         /// <param name="credentials"></param>
         /// <param name="config"></param>
         /// <returns></returns>
-        private static AmazonServiceClient CreateClient(Type serviceInterfaceType, AWSCredentials credentials, ClientConfig config)
+        private static AmazonServiceClient CreateClient(AWSCredentials credentials, ClientConfig config)
         {
-            var clientTypeName = serviceInterfaceType.Namespace + "." + serviceInterfaceType.Name.Substring(1) + "Client";
-            var clientType = serviceInterfaceType.GetTypeInfo().Assembly.GetType(clientTypeName);
+#if NET8_0_OR_GREATER
+            return T.CreateDefaultServiceClient(credentials, config) as AmazonServiceClient;
```

Comment:
This is where the new static interface methods that are added to the generator are called. Interesting thing about static interface methods especially with being declared as `abstract` is the type has to be known at compile time. That is what drove the change in this class to use generic `T` instead of passing in the `Type`.